### PR TITLE
fix(auth): remove username en/decode logic 

### PR DESCRIPTION
**Pull request by israx**
#### Description of changes

Currently the library create auth keys to store tokens and uses the `username` as identifier. E.g 
`CognitoIdentityServiceProvider.${userPoolClientId}.${username}.idToken`

This username is encoded when setting tokens and decoded when getting tokens. This works fine in `LocalStorage`. The issue happens in Cookies as `js-cookie`(the package used by the library) will encode the keys and if username is an `email` it will double encode the `@` symbol.

This removes the encoding logic and attaches the plain username in the keys.

**NOTE:**

This change when published to `console-preview` will log out any users that were authenticated with the encoded key format.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by elorzafe — APPROVED** [review]
Looks good,

Thanks @israx 🎖️

File: packages/auth/src/providers/cognito/utils/refreshAuthTokens.ts [hunk 1]
```diff
@@ -69,6 +69,6 @@ export const refreshAuthTokens: TokenRefresher = async ({
 		idToken,
 		clockDrift,
 		refreshToken: refreshTokenString,
-		username: `${accessToken.payload.username}`,
```
Comment (HuiSF) [inline packages/auth/src/providers/cognito/utils/refreshAuthTokens.ts:72]: Tested against the new backend setup, the auth keys are consistent with this fix 👍

File: packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts [hunk 2]
```diff
@@ -81,8 +81,7 @@ export class DefaultTokenStore implements AuthTokenStore {
 		assert(tokens !== undefined, TokenProviderErrorCode.InvalidAuthTokens);
 		await this.clearTokens();
 
-		const lastAuthUser =
-			(tokens.username && encodeURIComponent(tokens.username)) ?? 'username';
```
Comment (ashwinkumar6) [inline packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts:85]: QQ,
Do we also need to update anythere here
1. [signInWithRedirect1](https://github.com/aws-amplify/amplify-js/blob/550cc9dc27c0909bf73618d94b1b5264aa02404c/packages/auth/src/providers/cognito/apis/signInWithRedirect.ts#L216-L218)
2. [signInWithRedirect2](https://github.com/aws-amplify/amplify-js/blob/550cc9dc27c0909bf73618d94b1b5264aa02404c/packages/auth/src/providers/cognito/apis/signInWithRedirect.ts#L270-L272)

Comment (HuiSF) [reply packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts:85]: +1, do we need to fallback to `username`, if `username` is missing from token payload, isn't considered as an error?

Comment (israx) [reply packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts:85]: I don't think so. The username stored in `LastAuthUser` will still be consistent. I'll give it a test to confirm

Comment (israx) [reply packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts:85]: tested signInWithRedirect and seems working fine on my end

Comment (israx) [reply packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts:85]: Just check, it does default to 'username'

**Review by HuiSF — APPROVED** [review]

**Pull request merged**